### DISCUSSION TITLE
Confusion with type & customType fields for the @Property() decorator.

### DISCUSSION THREAD

**Discussion by nurdism** [mikro-orm/mikro-orm#2708] [Q&A] [answered]
I'm really confused on what `type` vs `customType` does on an Entity. (I'm on v5 rc2, this may just be a bug)

If I use the `type` field, it doesn't seem to affect the output of the table creation 
```ts
@Entity()
export class Ent extends BaseEntity<Ent, 'id'> {
  @PrimaryKey({ type: BigIntType })
  id: string
}
```
```sql
create table "ent" ("id" varchar(255) not null);
```

But if I use the `customType` field, it does affect the table creation 
```ts
@Entity()
export class Ent extends BaseEntity<Ent, 'id'> {
  @PrimaryKey({ customType: new BigIntType() })
  id: string
}
```
```sql
create table "ent" ("id" bigserial primary key);
```

The documentation is kind of unclear on what is the appropriate field to use, [this page](https://mikro-orm.io/docs/next/custom-types) shows some example usage, but uses the `type` field.

**Comment by B4nan** [comment, marked as answer]
This happens if you have misaligned ORM versions - different version installed for `core` and `cli` packages. You end up with multiple installations of `core` package, and therefore multiple `Type` classes which breaks the detection (`instanceof` now can return false even if its actually the right class, but from different export).
